modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3385,7 +3385,13 @@ def sort_columns_by_row_values(self, rows, ascending=True, **kwargs):
     # Cat operations
     def cat_codes(self):
         def func(df) -> np.ndarray:
-            ser = df.iloc[:, 0]
+            # `df` is supposed to be consisted of multiple partitions,
+            # which should be concatenated before applying a function.
+            # `pd.concat` doesn't preserve categorical dtype
+            # if the dfs have categorical columns
+            # so we intentionaly restore the right dtype.
+            # TODO: revert the change when https://github.com/pandas-dev/pandas/issues/51362 is fixed.
+            ser = df.astype("category", copy=False).iloc[:, 0]
```

Comment:
```suggestion
            ser := df.iloc[:, 0]
            if ser.dtype != "category":
                ser = ser.astype("category", copy=False)
```
- should be faster to cast a single column rather than the whole frame
- let's not do extra-casting if not necessary